src/Nest/Aggregations/Bucket/GeoTileGrid/GeoTileGridAggregation.cs

```diff
@@ -0,0 +1,91 @@
+using System;
+using System.Linq.Expressions;
+using System.Runtime.Serialization;
+using Elasticsearch.Net.Utf8Json;
+
+namespace Nest
+{
+	[InterfaceDataContract]
+	[ReadAs(typeof(GeoTileGridAggregation))]
+	public interface IGeoTileGridAggregation : IBucketAggregation
+	{
+		/// <summary>
+		/// The name of the field indexed with GeoPoints.
+		/// </summary>
+		[DataMember(Name ="field")]
+		Field Field { get; set; }
+
+		/// <summary>
+		/// The zoom of the key used to define cells/buckets in the results.
+		/// </summary>
+		[DataMember(Name ="precision")]
+		GeoTilePrecision? Precision { get; set; }
+
+		/// <summary>
+		/// To allow for more accurate counting of the top cells returned in the final result the aggregation.
+		/// </summary>
+		[DataMember(Name ="shard_size")]
+		int? ShardSize { get; set; }
+
+		/// <summary>
+		/// The maximum number of geohash buckets to return.
+		/// </summary>
+		[DataMember(Name ="size")]
+		int? Size { get; set; }
+	}
+
+	public class GeoTileGridAggregation : BucketAggregationBase, IGeoTileGridAggregation
+	{
+		internal GeoTileGridAggregation() { }
+
+		public GeoTileGridAggregation(string name) : base(name) { }
+
+		/// <inheritdoc />
+		public Field Field { get; set; }
+
+		/// <inheritdoc />
+		public GeoTilePrecision? Precision { get; set; }
+
+		/// <inheritdoc />
+		public int? ShardSize { get; set; }
+
+		/// <inheritdoc />
+		public int? Size { get; set; }
+
+		internal override void WrapInContainer(AggregationContainer c) => c.GeoTile = this;
+	}
+
+	public class GeoTileGridAggregationDescriptor<T>
+		: BucketAggregationDescriptorBase<GeoTileGridAggregationDescriptor<T>, IGeoTileGridAggregation, T>
+			, IGeoTileGridAggregation
+		where T : class
+	{
+		/// <inheritdoc cref="IGeoTileGridAggregation.Field" />
+		Field IGeoTileGridAggregation.Field { get; set; }
+
+		/// <inheritdoc cref="IGeoTileGridAggregation.Precision" />
+		GeoTilePrecision? IGeoTileGridAggregation.Precision { get; set; }
+
+		/// <inheritdoc cref="IGeoTileGridAggregation.ShardSize" />
+		int? IGeoTileGridAggregation.ShardSize { get; set; }
+
+		/// <inheritdoc cref="IGeoTileGridAggregation.Size" />
```

Comment:
`///inheritdoc` superfluous on explicit implemented properties (it already has this doc)